ceph/ceph_admin.py

```diff
@@ -548,8 +548,55 @@ def ceph_iscsi(self, nodes, **kwargs):
             nodes: iscsi node list
             kwargs: key-value arguments
         """
-        # todo: Handle iscsi
-        raise NotImplementedError
+        op = kwargs.get('op', 'create').lower()
+
+        nodes = nodes if isinstance(nodes, list) else [nodes]
+        if op == "create":
+            iscsis = []
+            daemons = []
+            for node in nodes:
+                logger.info("adding iscsi on %s-%s" % (node.hostname,
+                                                       node.ip_address))
+                iscsis.append("{host}=iscsi.{host}".format(host=node.hostname))
+                daemons.append("iscsi.{host}".format(host=node.hostname))
+
+            if iscsis:
+                pool_name = "iscsi"
+
+                self.shell(
+                    remote=self.installer,
+                    args=[
+                        'ceph', 'osd', 'pool', 'create',
+                        pool_name, '3', '3', 'replicated'
+                    ]
+                )
+
+                self.shell(
+                    remote=self.installer,
+                    args=[
+                        'ceph', 'osd', 'pool', 'application',
+                        'enable', pool_name, 'rbd'
+                    ]
+                )
+
+                self.shell(
+                    remote=self.installer,
+                    args=[
+                        'ceph', 'orch', 'apply', 'iscsi', pool_name,
+                        'user', 'password', '--placement',
+                        "'{};{}'".format(len(iscsis), ";".join(iscsis))
+                    ]
+                )
+
+            # check daemon existence
+            assert self.check_exist(
+                daemon='iscsi',
+                ids=daemons,
+                timeout=900,
+            )
+        elif op == "purge":
```

Comment:
Could we avoid the `elif` statement instead opt to use `else`?

Reply:
we may have update/modify in future to update services, we can take this in future if we don't have further actions on services.